llvm/lib/ProfileData/InstrProfWriter.cpp

```diff
@@ -612,10 +618,12 @@ Error InstrProfWriter::writeImpl(ProfOStream &OS) {
   if (static_cast<bool>(ProfileKind & InstrProfKind::MemProf)) {
     MemProfSectionStart = OS.tell();
 
-    if (auto E =
-            writeMemProf(OS, MemProfData, MemProfVersionRequested,
-                         MemProfFullSchema, std::move(DataAccessProfileData)))
+    // Get the finalized MemProf summary that was built when adding records.
+    auto MemProfSum = MemProfSumBuilder.getSummary();
```

Comment:
Inline this into the call if you choose to change `writeMemProf` to take unique_ptr.